src/base-element.js

```diff
@@ -369,6 +369,25 @@ export class BaseElement {
   }
 
   /**
+   * Subclasses can override this method to create an analytics config.
+   * @returns {?Promise<!JSONType>}
+   */
+  createSandboxAnalyticsConfigCallback() {
+    return null;
+  }
+
+  /**
+   * Called on the inserted sandbox analytics element is ready
+   * Subclasses can override this method.
+   */
+  onSandboxAnalyticsReadyCallback() {}
```

Comment:
return a promise is probably better, but it's even more confusing for element (like ad) that can relayout. Note: `onSandboxAnalyticsReadyCallback()` can be called multiple times based on the number of layout times of an element.